spring-integration-file/src/main/java/org/springframework/integration/file/config/AbstractRemoteFileOutboundGatewayParser.java

```diff
@@ -59,7 +59,11 @@ protected BeanDefinitionBuilder parseHandler(Element element, ParserContext pars
 		else {
 			builder.addConstructorArgValue(element.getAttribute("command"));
 			if (element.hasAttribute(EXPRESSION_ATTRIBUTE)) {
-				builder.addConstructorArgValue(element.getAttribute(EXPRESSION_ATTRIBUTE));
+				String expression = element.getAttribute(EXPRESSION_ATTRIBUTE);
+				if ("++xsd.expression.default++".equals(expression)) {
+					expression = null;
+				}
+				builder.addConstructorArgValue(expression);
```

Comment:
Why just this not enough?
I mean we just call `builder.addConstructorArgValue(element.getAttribute(EXPRESSION_ATTRIBUTE));`.
Nothing more. 
What am I missing with `null` propagation and defering logic into the `AbstractRemoteFileOutboundGateway`.
Probably that point is that `expression` was not optional in the past, so we left that `if (element.hasAttribute(EXPRESSION_ATTRIBUTE)) {` logic in the parser.
But it turns out that we just need to always call `builder.addConstructorArgValue(expression);`, so we got the proper ctor resolution even if `expression` is `null`.